RUFAS/biophysical/manure/separators/separator.py

```diff
@@ -0,0 +1,277 @@
+        The efficiency of the separator in removing volatile solids from the manure.
+    total_solids_efficiency : float
+        The efficiency of the separator in removing total solids from the manure.
+    om : OutputManager
+        The output manager instance used to store and manage output data.
+
+    """
+
+    def __init__(
+        self,
+        name: str = "",
+        percent_dry_solids: float = 0.0,
+        ammoniacal_nitrogen_efficiency: float = 0.0,
+        nitrogen_efficiency: float = 0.0,
+        phosphorus_efficiency: float = 0.0,
+        potassium_efficiency: float = 0.0,
+        ash_efficiency: float = 0.0,
+        volatile_solids_efficiency: float = 0.0,
+        total_solids_efficiency: float = 0.0,
+    ) -> None:
+        """Initializes a new Separator."""
+        super().__init__(name=name, is_housing_emissions_calculator=False)
+        self._name: str = name
+        self._prefix: str = f"{self.__class__.__name__}"
+        self.held_manure: ManureStream | None = None
+        self.percent_dry_solids: float = percent_dry_solids
+        self.ammoniacal_nitrogen_efficiency: float = ammoniacal_nitrogen_efficiency
+        self.nitrogen_efficiency: float = nitrogen_efficiency
+        self.phosphorus_efficiency: float = phosphorus_efficiency
+        self.potassium_efficiency: float = potassium_efficiency
+        self.ash_efficiency: float = ash_efficiency
+        self.volatile_solids_efficiency: float = volatile_solids_efficiency
+        self.total_solids_efficiency: float = total_solids_efficiency
+        self.om = OutputManager()
+
+    def receive_manure(self, manure: ManureStream) -> None:
+        """
+        Takes in manure to be processed.
+
+        Parameters
+        ----------
+        manure : ManureStream
+            The manure to be processed.
+
+        """
+        if self.held_manure is None:
+            self.held_manure = manure
+        else:
+            self.held_manure += manure
+
+    def process_manure(self, conditions: CurrentDayConditions, time: Time) -> dict[str, ManureStream]:
+        """
+        Executes the daily manure processing operations.
+        For this class, it separates a portion of solids from the manure and returns the solid and liquid portions.
+
+        Parameters
+        ----------
+        conditions : CurrentDayConditions
+            Current weather and environmental conditions that manure is being processed in.
+        time : Time
+            Time instance containing the simulations temporal information.
+
+        Returns
+        -------
+        dict[str, ManureStream]
+            A dictionary containing:
+            - "solid" : ManureStream
+                The solid portion of the separated manure.
+            - "liquid" : ManureStream
+                The liquid portion of the separated manure.
+
+        """
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self.process_manure.__name__,
+        }
```

Comment:
```suggestion
        info_map = {
            "class": self.__class__.__name__,
            "function": self.process_manure.__name__,
            "prefix": self._prefix,
        }
```
I think the prefix should be set in the info map. This will be the difference between `ManureSeparator.SeparatedSolids.YourProcessorNameHere.water` and `Separator.process_manure.ManureSeparator.SeparatedSolids.YourProcessorNameHere.water`, and I interpreted the [output specification](https://docs.google.com/spreadsheets/d/1LTLLNjA580pkIWm6ILqwrCDMb9903NnIV6GAw0W6l_8/edit?gid=0#gid=0) as asking for the former.

Also, I know that the output specification calls for `ManureSeparator.SeparatedSolids.YourProcessorNameHere.water`, but would it possible to do `ManureSeparator.YourProcessorNameHere.SeparatedSolids.water` instead? The only benefit is that this would make the variable name more uniform with the outputs from other Processors because it would be following the prefix set in `_prefix` in `Processor`.

@elle-andreen could you please weigh in on these?

Reply:
That's fine with me! I wrote up the output names long ago and they're open to modification for clarity or ease of implementation